[CI] CrossClusterQueryWithPartialResultsIT testFailToStartRequestOnRemoteCluster failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #9903 / windows-2019_checkpart3_platform-support-windows](https://gradle-enterprise.elastic.co/s/fgzxk4i52ix7i)
- [elasticsearch-periodic-platform-support #9896 / oraclelinux-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/nf5txunhwrqw4)
- [elasticsearch-periodic-platform-support #9875 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/yder2obdey52e)
- [elasticsearch-periodic-platform-support #9875 / oraclelinux-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/ivh2nocckm4q6)
- [elasticsearch-periodic-platform-support #9854 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/vlws7pxrfzytq)
- [elasticsearch-periodic-platform-support #9826 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/qe3scoe3ms2cs)
- [elasticsearch-periodic-platform-support #9826 / almalinux-8-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/4mkuabyh2kxnq)
- [elasticsearch-periodic-platform-support #9791 / sles-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/7cnrgcccrosva)
- [elasticsearch-periodic-platform-support #9777 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/iaqrxkhjpmgmw)
- [elasticsearch-periodic-platform-support #9770 / sles-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/uyh4zqrfkdjqq)

**Reproduction Line:**
```
gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.CrossClusterQueryWithPartialResultsIT.testFailToStartRequestOnRemoteCluster" -Dtests.seed=7F336E410AE7D9CD -Dtests.locale=za -Dtests.timezone=America/Buenos_Aires -Druntime.java=24
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.CrossClusterQueryWithPartialResultsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testFailToStartRequestOnRemoteCluster'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <skipped>
     but: was <partial>
```

**Issue Reasons:**
- [8.19] 11 failures in test testFailToStartRequestOnRemoteCluster (1.7% fail rate in 651 executions)
- [8.19] 2 failures in step oraclelinux-8_platform-support-unix (10.0% fail rate in 20 executions)
- [8.19] 2 failures in step sles-12_platform-support-unix (9.5% fail rate in 21 executions)
- [8.19] 2 failures in step oraclelinux-7_platform-support-unix (10.5% fail rate in 19 executions)
- [8.19] 9 failures in pipeline elasticsearch-periodic-platform-support (40.9% fail rate in 22 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -307,9 +307,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.plugin.DataNodeRequestSenderIT
   method: testSearchWhileRelocating
   issue: https://github.com/elastic/elasticsearch/issues/128500
-- class: org.elasticsearch.xpack.esql.action.CrossClusterQueryWithPartialResultsIT
-  method: testFailToStartRequestOnRemoteCluster
-  issue: https://github.com/elastic/elasticsearch/issues/128545
 - class: org.elasticsearch.compute.operator.LimitOperatorTests
   method: testEarlyTermination
   issue: https://github.com/elastic/elasticsearch/issues/128721

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterQueryWithPartialResultsIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterQueryWithPartialResultsIT.java
@@ -44,6 +44,7 @@ import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.lessThanOrEqualTo;
 import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.oneOf;
 
 public class CrossClusterQueryWithPartialResultsIT extends AbstractCrossClusterTestCase {
 
@@ -291,7 +292,11 @@ public class CrossClusterQueryWithPartialResultsIT extends AbstractCrossClusterT
                 assertThat(returnedIds, equalTo(local.okIds));
                 assertClusterSuccess(resp, LOCAL_CLUSTER, local.okShards);
                 EsqlExecutionInfo.Cluster remoteInfo = resp.getExecutionInfo().getCluster(REMOTE_CLUSTER_1);
-                assertThat(remoteInfo.getStatus(), equalTo(EsqlExecutionInfo.Cluster.Status.SKIPPED));
+                // It could also return partial on failure
+                assertThat(
+                    remoteInfo.getStatus(),
+                    oneOf(EsqlExecutionInfo.Cluster.Status.SKIPPED, EsqlExecutionInfo.Cluster.Status.PARTIAL)
+                );
                 assertClusterFailure(resp, REMOTE_CLUSTER_1, simulatedFailure.getMessage());
             }
         } finally {

